cohorts/patient.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) 2016. Mount Sinai School of Medicine
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+from .utils import require_id_str
+
+class Patient(object):
+    """
+    Represents a patient, which contains zero or more of: normal `Sample`,
+    tumor `Sample`.
+
+    Parameters
+    __________
+    id : str
+        ID of the patient.
+    os : int
+        Overall survival in days.
+    pfs : int
+        Progression-free survival in days.
+    deceased : bool
+        Is the patient deceased?
+    progressed : bool
+        Has the patient progressed?
+    progressed_or_deceased : bool
+        Has the patient either progressed or passed away?
+    benefit : bool
+        Has the patient seen a durable clinical benefit?
+    snv_vcf_paths : list
+        List of paths to SNV VCFs or this patient; multple VCFs get merged.
+    indel_vcf_paths : list
+        List of paths to indel VCFs for this patient; multple VCFs get merged.
+    normal_sample : Sample
+        This patient's normal `Sample`.
+    tumor_sample: Sample
+        This patient's tumor `Sample`.
+    hla_alleles : list
+        A list of this patient's HLA class I alleles.
+    additional_data : dict
+        A dictionary of additional data: name of datum mapping to value.
+    """
+    def __init__(self,
+                 id,
+                 os,
+                 pfs,
+                 deceased,
+                 progressed=None,
+                 progressed_or_deceased=None,
+                 benefit=None,
+                 snv_vcf_paths=[],
+                 indel_vcf_paths=[],
+                 normal_sample=None,
+                 tumor_sample=None,
+                 hla_alleles=None,
+                 additional_data=None,
+                 cohort=None):
+        require_id_str(id)
+        self.id = id
```

Comment:
Not necessary here, but saw a nice library to simplify things like this https://attrs.readthedocs.io/en/stable/examples.html